src/backend/distributed/planner/recursive_planning.c

```diff
@@ -206,6 +212,11 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 	/* descend into subqueries */
 	query_tree_walker(query, RecursivelyPlanSubqueryWalker, context, 0);
 
+	if (ShouldRecursivelyPlanSetOperation(query, context))
```

Comment:
Would be nice to have some comment after `query_tree_walker ()` that explains when we hit here? It might not be obvious for the readers